src/Microsoft.TestPlatform.AdapterUtilities/ManagedNameUtilities/ManagedNameHelper.Reflection.cs

```diff
@@ -260,16 +256,17 @@ bool Filter(MemberInfo mbr, object param)
         methods = type.GetRuntimeMethods().Where(m => Filter(m, null)).ToArray();
 #endif
 
-#if NET20
-        return (MethodInfo)SingleOrDefault(methods);
-#else
-        return (MethodInfo)methods.SingleOrDefault();
-#endif
+        return (MethodInfo?)methods.SingleOrDefault();
     }
 
-    private static int[] AppendTypeString(StringBuilder b, Type type, bool closedType)
+    private static int[]? AppendTypeString(StringBuilder b, Type? type, bool closedType)
     {
-        int[] hierarchies = null;
+        int[]? hierarchies = null;
+
+        if (type is null)
```

Comment:
Not clear what to do when `type` is null.

Reply:
We should return `null` in that case. This method might be called recursively until we reach the end of type tree. We stop when it's null.